pkg/framework/framework.go

```diff
@@ -176,3 +177,26 @@ func WaitForValidatingWebhook(client runtimeclient.Client, name string) bool {
 
 	return true
 }
+
+// WaitForEvent expects to find the given event
+func WaitForEvent(c runtimeclient.Client, kind, name, reason string) error {
+	return wait.PollImmediate(RetryMedium, WaitMedium, func() (bool, error) {
+		eventList := corev1.EventList{}
+		if err := c.List(context.TODO(), &eventList); err != nil {
```

Comment:
Use background for tests where we don't think we will replace it at a later date
```suggestion
		if err := c.List(context.Background(), &eventList); err != nil {
```